tests/tests_pytorch/models/test_amp.py

```diff
@@ -23,9 +24,26 @@
 from lightning_lite.plugins.environments import SLURMEnvironment
 from pytorch_lightning import Trainer
 from pytorch_lightning.demos.boring_classes import BoringModel, RandomDataset
+from pytorch_lightning.utilities.imports import _APEX_AVAILABLE
 from tests_pytorch.helpers.runif import RunIf
 
 
+@contextmanager
+def _suppress_apex_deprecation():
+    if not _APEX_AVAILABLE:
+        yield
+        return
+
+    import apex
+
+    if not hasattr(apex, "DeprecatedFeatureWarning"):
+        yield
+        return
+
+    with pytest.warns(apex.DeprecatedFeatureWarning, match="apex.amp is deprecated"):
+        yield
+
+
```

Comment:
This could be a `pytest.fixture(autouse=True)`

Reply:
No, because in some tests we need to conditionally activate it based on parameterizations. 
Feel free to commit something if you want, but I think I won't go deeper myself.